### ISSUE TITLE
The oldcmarkup extension doesn't work with default_role config option

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#487]
When using old C markup and defining 

```
default_role = 'cfunc'
```

in conf.py, Sphinx crashes when it encounters text that uses the default interpreted text role.

The problem is in the `sphinx.ext.oldcmarkup` extension. The `old_crole` function gets an empty string as the `typ` parameter, which causes the call to `env.lookup_domain_element()` fail.

I managed to get it to work by changing the `old_crole` function a bit:

```
if not typ:
    typ = env.config.default_role
```

I'm not sure if this is a good fix, but at least it works.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/487
- Originally reported by: Anonymous
- Originally created at: 2010-08-05T11:30:26.543

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2010-08-05 10:06:47+00:00_

The suggested fix is correct AFAICS, applied in <<cset eedef65aa6d3673b81ac6fcd6c9b05ec74a2567e>>.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-10-18 07:29:32+00:00_

Removing version: 1.0.1 (automated comment)
